lib/config/set-envs.js

```diff
@@ -8,11 +8,14 @@
 // and the value is a string or array.  Otherwise return false.
 const envKey = (key, val) => {
   return !/^[/@_]/.test(key) &&
-    (typeof val === 'string' || Array.isArray(val)) &&
-    `npm_config_${key.replace(/-/g, '_').toLowerCase()}`
+    (typeof envVal(val) === 'string') &&
+      `npm_config_${key.replace(/-/g, '_').toLowerCase()}`
 }
 
-const envVal = val => Array.isArray(val) ? val.join('\n\n') : val
+const envVal = val => Array.isArray(val) ? val.map(v => envVal(v)).join('\n\n')
+  : val === null || val === undefined || val === false ? ''
```

Comment:
o_O can we at least start nesting these? ex.
```
const envVal = val => Array.isArray(val) ? val.map(v => envVal(v)).join('\n\n')
  : val === null || val === undefined || val === false ? ''
    : typeof val === 'object' ? null
      : String(val)
```

Reply:
We've moved to lining up the `:` for easier scanning down a straight left-edge, so the CLI matches all the other repos.